iot/manager/manager.js

```diff
@@ -744,23 +746,29 @@ function getRegistry (client, registryId, projectId, cloudRegion) {
 // Returns an authorized API client by discovering the Cloud IoT Core API with
 // the provided API key.
 function getClient (serviceAccountJson, cb) {
-  const serviceAccount = JSON.parse(fs.readFileSync(serviceAccountJson));
-  const jwtAccess = new google.auth.JWT();
-  jwtAccess.fromJSON(serviceAccount);
-  // Note that if you require additional scopes, they should be specified as a
-  // string, separated by spaces.
-  jwtAccess.scopes = 'https://www.googleapis.com/auth/cloud-platform';
-  // Set the default authentication to the above JWT access.
-  google.options({ auth: jwtAccess });
-
-  const discoveryUrl = `${DISCOVERY_API}?version=${API_VERSION}`;
-
-  google.discoverAPI(discoveryUrl, {}, (err, client) => {
-    if (err) {
-      console.log('Error during API discovery', err);
-      return undefined;
+  google.auth.getClient().then(authClient => {
+    const discoveryUrl =
+        `${DISCOVERY_API}?version=${API_VERSION}`;
+
+    if (authClient.createScopedRequired && authClient.createScopedRequired()) {
```

Comment:
You don't need to do this :) Just pass the list of scopes into the `getClient` method:

```js
google.auth.getClient({
  scopes: ['https://www.googleapis.com/auth/cloud-platform']
}).then(auth => {
  google.options({ auth });
  google.discoverAPI(....
});
```